modules/standard/DateTime.chpl

```diff
@@ -1197,7 +1197,36 @@ module DateTime {
     timeStruct.tm_wday = (weekday(): int(32) + 1) % 7; // shift Sunday to 0
     timeStruct.tm_yday = (this.replace(tzinfo=nil) - new datetime(year, 1, 1)).days: int(32);
 
-    strftime(c_ptrTo(buf), bufLen, fmt.c_str(), timeStruct);
+    // Iterate over format specifiers in strftime()
+    iter strftok(s)
```

Comment:
Could this use `string.replace()` instead of walking the characters?

```chapel
const micro = "%06i".format(chpl_time.chpl_microsecond);
const refmt = s.replace("%f", micro);
```

Reply:
> Could this use `string.replace()` instead of walking the characters?
> 
> ```chapel
> const micro = "%06i".format(chpl_time.chpl_microsecond);
> const refmt = s.replace("%f", micro);
> ```

No, because `%%f` needs to be printed as `%f` instead of as `%012345`.

I added a test case with `%%f` to test this case:
```chapel
assert(t.strftime("%m %d %y %S %M %H %%f %j") == "12 31 04 33 22 06 %f 366");
```